# Exercise 1.05
## Comparison to the baseline

In this exercise we will put the model performance into context. The accuracy we attained from our model seemed good, but we need something to compare it to. Since machine learning model performnace is relative it is important to develop a robust baseline with which to compare models to.

A baseline model should be simple and well understood procedure, and the performance of this model should act as the bedrock for the lowest acceptable performance for any model we build.

For classification models a useful and easy baseline model is to calculate the mode outcome value. For example, in our example if there are 60% false values, our baseline model would be to predict false for every value, which would give us an accuracy of 60%.

Let's use this method for our baseline

In [1]:
import pandas as pd
target = pd.read_csv('../data/OSI_target_e2.csv')

Lets caluclate the proportion of each target value

In [2]:
target['Revenue'].value_counts()/target.shape[0]*100

0    84.525547
1    15.474453
Name: Revenue, dtype: float64

Now we can see that our baseline model would to predict '0', and that this with a 84.525547% accuracy.

In [3]:
y_baseline = pd.Series(data=[0]*target.shape[0])

We can also look at the other performance metrics

In [4]:
from sklearn import metrics
precision, recall, fscore, _ = metrics.precision_recall_fscore_support(y_pred=y_baseline, y_true=target['Revenue'], average='macro', zero_division=1)
print(f'Precision: {precision:.4f}\nRecall: {recall:.4f}\nfscore: {fscore:.4f}')

Precision: 0.9226
Recall: 0.5000
fscore: 0.4581


These we be the values in which we compare any models we train against